app/cli/internal/telemetry/telemetry.go

```diff
@@ -49,28 +50,57 @@ func NewCommandTracker(client Client) *CommandTracker {
 
 // Track sends a command event to the telemetry.Client.
 func (t *CommandTracker) Track(ctx context.Context, cmd string, tags Tags) error {
+	// Ensure a valid client is available.
 	if t.client == nil {
 		return nil
 	}
 
-	computedTags := loadDefaultTags()
-	// Add on top of the default tags the ones passed as argument.
-	// TODO: Add a way to avoid the override default tags.
-	for k, v := range tags {
-		computedTags[k] = v
-	}
+	// Load default tags and merge with user-provided tags.
+	computedTags := mergeTags(loadDefaultTags(), tags)
+
+	// Set the command tag.
 	computedTags["command"] = cmd
 
-	var id string
-	id, err := machineid.ProtectedID("chainloop")
-	if err != nil {
-		// If the machine ID is not available, use a default ID otherwise the underlying library will fail.
-		id = "default-id"
-	}
+	// Determine the user ID.
+	id := DetermineUserID(computedTags)
 
+	// Track the event with computed tags.
 	return t.client.TrackEvent(ctx, commandTrackerEventName, id, computedTags)
 }
 
+// Merges two tag maps.
+func mergeTags(defaultTags, userTags Tags) Tags {
+	result := make(Tags)
+	for k, v := range defaultTags {
+		result[k] = v
+	}
+	for k, v := range userTags {
+		result[k] = v
+	}
+	return result
+}
+
+// DetermineUserID Determines the user ID using available information or generates a random one.
+func DetermineUserID(tags Tags) string {
```

Comment:
this function probably doesn't need to be exposed. I am assuming it is because the test package uses `_test`. In that case I'd ask the question if we need the `test` file to use a different package or just treat it as unit tests in the `telemetry` package.

What do you think?